packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/integration/readwise/connect.tsx

```diff
@@ -0,0 +1,164 @@
+import { Button, Input, Modal, notify } from '@affine/component';
+import { useAsyncCallback } from '@affine/core/components/hooks/affine-async-hooks';
+import { IntegrationService } from '@affine/core/modules/integration';
+import { Trans, useI18n } from '@affine/i18n';
+import { ReadwiseLogoDuotoneIcon } from '@blocksuite/icons/rc';
+import { useService } from '@toeverything/infra';
+import { type FormEvent, useCallback, useState } from 'react';
+
+import { IntegrationCardIcon } from '../card';
+import {
+  actionButton,
+  connectDesc,
+  connectDialog,
+  connectFooter,
+  connectInput,
+  connectTitle,
+  getTokenLink,
+  inputErrorMsg,
+} from './index.css';
+
+const ConnectDialog = ({
+  onClose,
+}: {
+  onClose: (successToken?: string) => void;
+}) => {
+  const t = useI18n();
+  const [status, setStatus] = useState<'idle' | 'verifying' | 'error'>('idle');
+  const [token, setToken] = useState('');
+  const readwise = useService(IntegrationService).readwise;
+
+  const handleCancel = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  const onOpenChange = useCallback(
+    (open: boolean) => {
+      if (!open) handleCancel();
+    },
+    [handleCancel]
+  );
+
+  const handleInput = useCallback((e: FormEvent<HTMLInputElement>) => {
+    setToken(e.currentTarget.value);
+    setStatus('idle');
+  }, []);
+
+  const handleResult = useCallback(
+    (success: boolean, token: string) => {
+      if (success) {
+        onClose(token);
```

Comment:
onClose is not necessary?

```suggestion
        readwise.updateSetting('token', token);
```